pkl-core/src/main/java/org/pkl/core/runtime/AnsiFormatter.java

```diff
@@ -0,0 +1,144 @@
+/*
+ * Copyright © 2024 Apple Inc. and the Pkl project authors. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.pkl.core.runtime;
+
+/*
+   TODO:
+     * Manage "active state" to reduce "escape code spew" in output
+*/
```

Comment:
Given the current API, not sure if this is possible?

Reply:
It was (`fg` and `fgBright` could just check current state and emit an escape or not), but I think the new approach covers this also.